_posts/2021-02-08-ill-be-back.md

```diff
@@ -0,0 +1,60 @@
+---
+layout: post
+header_image: /img/posts/ill-be-back/header.jpg
+title: "I'll be back"
+author: Mauro Servienti
+synopsis: ""
+tags:
+- Distributed Systems
+- Timeouts
+- Long running
+- Sagas
+- Delayed messages
+---
+
+Time, as a concept, is both elementary and incredibly complex. Up to the point that physicians like Carlo Rovelli say that [it doesn't exist](https://youtu.be/vgsoI4ZUkUA). If you spend some time, ironically, time exists if you can spend it, diving into the theories' nuances, it's clear that time exists at a particular scale, it's not unique and ubiquitous, and it can be stretched and manipulated. And yes, at the quantum scale, it doesn't exist. Mind-blowing and fascinating.
+
+If we look at time from the perspective of systems design, it has many nuances and complexities.
+On the one hand, there is clock drift: different machines have different views of time; it's not unique and ubiquitous. On the other hand, there is the conceptual complexity of modeling the passage of time. Modeling the passage of time as a clock does seems to be a natural thing to do. It translates well the [user mental model]() into the system design and provides a reliable solution. However, if we look deeper, it might not be so good as a solution.
+
+## Clocks are batch jobs
+
+Time ticks, human beings observe the ticking of time and react.
+
+Imagining a clock scale of 1 day, in which clocks will only tell which day of the week is today. We will be in a situation where there is a long list of things to do today and someone that picks from the list and executes the corresponding task. It'll be a big mess, 24 hours are quite a long time, and those to-do lists can become quite long. It's very tempting to increase the scale and make it so that a clock ticks every one hour instead of every 24. Is anything changed? Might be, we might be lucky and discover that the granularity of a one-hour scale in our context is better. Regardless of the clock scale of choice, we're approaching the problem with a batch job oriented solution. We'll be sitting there waiting for a clock tick to kick off an allocated slot to perform some work regardless of the amount of work.
```

Comment:
```suggestion
Imagining a clock scale of 1 day, in which clocks will only tell which day of the week is today. We will be in a situation where there is a long list of things to do today and someone that picks from the list and executes the corresponding task. It'll be a big mess, 24 hours is quite a long chunk of time, and those to-do lists can become quite long. It's very tempting to increase the scale and make it so that a clock ticks every hour instead of every 24. Has anything changed? Maybe, we might be lucky and discover that the granularity of a one-hour scale in our context is better. Regardless of the clock scale of choice, we're approaching the problem with a batch job oriented solution. We'll be sitting there waiting for a clock tick to kick off an allocated slot to perform some work regardless of the amount of work.
```